src/main/kotlin/no/nav/familie/ef/sak/vedtak/historikk/VedtakHistorikkService.kt

```diff
@@ -77,6 +126,61 @@ class VedtakHistorikkService(
             }
     }
 
+    private fun mapUtgifterBarnetilsyn(
+        historikk: List<AndelHistorikkDto>,
+        fra: YearMonth,
+        beløp: (AndelMedGrunnlagDto) -> Int
+    ): List<PeriodeMedBeløpDto> {
+        return historikk
+            .filter { beløp(it.andel) > 0 }
+            .slåSammen { a, b ->
+                sammenhengende(a, b) &&
+                    beløp(a.andel) == beløp(b.andel)
+            }
+            .fraDato(fra)
+            .map {
+                PeriodeMedBeløpDto(
+                    årMånedFra = it.andel.periode.fom,
+                    årMånedTil = it.andel.periode.tom,
+                    periode = it.andel.periode,
+                    beløp(it.andel)
+                )
+            }
+    }
+
+    private fun mapBarnetilsynPerioder(
+        historikk: List<AndelHistorikkDto>,
+        fra: YearMonth,
+        behandlingId: UUID
+    ): List<UtgiftsperiodeDto> {
+        val barnMap = mapHistoriskeBarn(behandlingId, historikk)
+        return historikk
+            .slåSammen { a, b ->
+                sammenhengende(a, b) &&
+                    a.andel.barn.toSet() == b.andel.barn.toSet() &&
+                    a.andel.utgifter.compareTo(b.andel.utgifter) == 0
+            }
+            .fraDato(fra)
+            .map {
+                UtgiftsperiodeDto(
+                    årMånedFra = it.andel.periode.fom,
+                    årMånedTil = it.andel.periode.tom,
+                    periode = it.andel.periode,
+                    barn = it.andel.barn.map { barnId -> barnMap[barnId] ?: error("Fant ikke match for barn=$barnId") },
+                    utgifter = it.andel.utgifter.toInt(),
+                    erMidlertidigOpphør = false
```

Comment:
Hvordan løses egentlig denne? erMidlertidigOpphør = false

Reply:
Historikken inneholder ikke midlertidig opphør
Tror det finnes en annen favro-oppgave på å få lagt til det for å få vist det i vedtakshistorikken, då blir denne endret